src/main/scala/org/allenai/pipeline/Producer.scala

```diff
@@ -125,7 +156,7 @@ trait CachingDisabled extends CachingEnabled {
   override def cachingEnabled: Boolean = false
 }
 
-class PersistedProducer[T, -A <: Artifact](
+class PersistedProducer[T, -A <: Artifact] private[pipeline] (
```

Comment:
The `Producer.persist` method returns a `PersistedProducer` not a simple `Producer` This really should be changed to return an interface instead of a concrete class.  The only additional thing that `PersistedProducer` exposes is the `Artifact` that the data was saved to.